test/letters/containers/AddressSection.unit.spec.jsx

```diff
@@ -141,4 +159,156 @@ describe('<AddressSection>', () => {
     expect(ReactTestUtils.scryRenderedDOMComponentsWithTag(component, 'select')).to.be.empty;
   });
 
+  // NOTE: This is a bit of a misnomer; it only tests if countries are unavailable, but that should be sufficient
+  it('should show addressUpdateUnavailable if countries or states lists aren\'t available', () => {
+    const props = Object.assign({}, defaultProps, { countriesAvailable: false });
+    const component = ReactTestUtils.renderIntoDocument(<AddressSection {...props}/>);
+    const tree = getFormDOM(component);
+
+    // Start editing
+    tree.click('button.usa-button-outline');
+    expect(tree.getElement('.usa-alert-heading').textContent).to.contain('Address update unavailable');
+  });
+
+  it('should load address in new props after mounting', () => {
+    const props = cloneDeep(defaultProps);
+    props.savedAddress = {};
+    const tree = SkinDeep.shallowRender(<AddressSection {...props}/>);
+
+    const instance = tree.getMountedInstance();
+    expect(instance.state.editableAddress).to.equal(props.savedAddress);
+
+    const newProps = Object.assign({}, props, { savedAddress: { addressOne: '123 Main St' } });
+    instance.componentWillReceiveProps(newProps);
+    expect(instance.state.editableAddress).to.equal(newProps.savedAddress);
+  });
+
+  it('should not call saveAddress when Cancel is clicked', () => {
```

Comment:
Do you think we need this test? I guess you could say that we're proactively preventing someone from breaking things in the future if they don't understand the difference between saved & editable addresses and decide to always call saveAddress?

Reply:
Yeah, basically. Just making sure that the button does what we expect it to for future-proofing. :man_shrugging:

Maybe overkill, but I like being thorough.